geth/api/app_state.go

```diff
@@ -0,0 +1,39 @@
+package api
+
+import (
+	"fmt"
+	"strings"
+)
+
+// AppState represents if an app is in foreground or not,
+// see https://facebook.github.io/react-native/docs/appstate.html
+type AppState string
+
+func (a AppState) String() string {
+	return string(a)
+}
+
+const (
+	AppStateForeground = AppState("foreground")
+	AppStateBackground = AppState("background")
+	AppStateInactive   = AppState("inactive")
+
+	AppStateInvalid = AppState("")
+)
+
+func validAppStates() []AppState {
```

Comment:
Why no constant here?

Reply:
basically to make it _truly_ constant.
Arrays can only be declared with `var` and it is mutable.
No big deal though, I can change it.

Reply:
Expected it and yeah, sadly no real constant slices. In global cases I would do it similar.

Reply:
So, would you mind me keeping it as is or do you want me to make it a `var`?

Reply:
Just keep it, it's more robust and elegant. Maybe a little comment. 

```
// validAppStates returns an immutable set of valid states.
```

Reply:
done!